lib/rewrite/tests/test_set_severity.c

```diff
@@ -0,0 +1,136 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <criterion/criterion.h>
+
+#include "apphook.h"
+#include "rewrite/rewrite-set-severity.h"
+#include "logmsg/logmsg.h"
+#include "grab-logging.h"
+
+GlobalConfig *cfg = NULL;
+
+static LogTemplate *
+_create_template(const gchar *str)
+{
+  GError *error = NULL;
+  LogTemplate *template = log_template_new(cfg, NULL);
+  log_template_compile(template, str, &error);
+
+  cr_expect_null(error);
+
+  return template;
+}
+
+Test(set_severity, text)
+{
+  LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+  LogMessage *msg = log_msg_new_empty();
+
+  LogRewrite *set_severity = log_rewrite_set_severity_new(_create_template("error"), cfg);
+
+  log_pipe_init(&set_severity->super);
+  log_msg_ref(msg);
+  log_pipe_queue(&set_severity->super, msg, &path_options);
+
+  cr_assert_eq(msg->pri & LOG_PRIMASK, 3);
+
+  log_msg_unref(msg);
+  log_pipe_deinit(&set_severity->super);
+  log_pipe_unref(&set_severity->super);
+}
+
+Test(set_severity, numeric)
+{
+  LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+  LogMessage *msg = log_msg_new_empty();
+
+  LogRewrite *set_severity = log_rewrite_set_severity_new(_create_template("1"), cfg);
+
+  log_pipe_init(&set_severity->super);
+  log_msg_ref(msg);
+  log_pipe_queue(&set_severity->super, msg, &path_options);
+
+  cr_assert_eq(msg->pri & LOG_PRIMASK, 1);
+
+  log_msg_unref(msg);
+  log_pipe_deinit(&set_severity->super);
+  log_pipe_unref(&set_severity->super);
+}
+
+Test(set_severity, large_number)
+{
+  debug_flag = 1;
+
+  LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+  LogMessage *msg = log_msg_new_empty();
+
+  LogRewrite *set_severity = log_rewrite_set_severity_new(_create_template("8"), cfg);
+
+  log_pipe_init(&set_severity->super);
+  log_pipe_queue(&set_severity->super, msg, &path_options);
+
+  assert_grabbed_log_contains("invalid severity to set");
+
+  log_pipe_deinit(&set_severity->super);
+  log_pipe_unref(&set_severity->super);
+}
+
+Test(set_severity, invalid)
+{
+  debug_flag = 1;
+
+  LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+  LogMessage *msg = log_msg_new_empty();
+
+  LogRewrite *set_severity = log_rewrite_set_severity_new(_create_template("random-text"), cfg);
+
+  log_pipe_init(&set_severity->super);
+  log_pipe_queue(&set_severity->super, msg, &path_options);
+
+  assert_grabbed_log_contains("invalid severity to set");
```

Comment:
What do you think about showing the behavior in case the rewrite rule failed to set the severity (additionally to the error message) ?
Either check explicitly to the expected default value (LOG_NOTICE) or get the current severity before `log_pipe_queue` is called and then check that it isn't changed?

Example about the former:
```
+++ b/lib/rewrite/tests/test_set_severity.c
@@ -85,12 +85,14 @@ Test(set_severity, large_number)
 
   LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
   LogMessage *msg = log_msg_new_empty();
+  cr_assert_eq(msg->pri & LOG_PRIMASK, LOG_NOTICE);
 
   LogRewrite *set_severity = log_rewrite_set_severity_new(_create_template("8"), cfg);
 
   log_pipe_init(&set_severity->super);
   log_pipe_queue(&set_severity->super, msg, &path_options);
 
+  cr_assert_eq(msg->pri & LOG_PRIMASK, LOG_NOTICE);
   assert_grabbed_log_contains("invalid severity to set");
```

Reply:
Makes sense!